### PR TITLE
Fixed deferring of closing dbs

### PR DESCRIPTION
**Pull request by rkapka**
**What type of PR is this?**

Bug fix

**What does this PR do? Why is it needed?**

Previous implementation of merging validators had a bug when closing open connections. The `defer` statement was declared not early enough. This PR fixes the problem.

**Which issues(s) does this PR fix?**

N/A

**Other notes for review**

I am not sure if my solution is idiomatic Go, but it does the job 😉

### CODE REVIEW

File: validator/accounts/account.go [hunk 1]
```diff
@@ -223,7 +223,15 @@ func HandleEmptyKeystoreFlags(cliCtx *cli.Context, confirmPassword bool) (string
 
 // Merge merges data from validator databases in sourceDirectories into a new store, which is created in targetDirectory.
 func Merge(ctx context.Context, sourceDirectories []string, targetDirectory string) error {
-	var sourceStores []*db.Store
+	var sourceStoreSlice = make([]*db.Store, 0)
+	var sourceStores = &sourceStoreSlice
```
Comment (terencechain) [inline validator/accounts/account.go:227]: why do we need to deference `sourceStoreSlice`?

File: validator/accounts/account.go [hunk 2]
```diff
@@ -223,7 +223,15 @@ func HandleEmptyKeystoreFlags(cliCtx *cli.Context, confirmPassword bool) (string
...
+	defer func() {
+		for _, store := range *sourceStores {
```
Comment (terencechain) [inline validator/accounts/account.go:229]: why can't `sourceStoreSlice` work here?

File: validator/accounts/account.go (hunk 1 above)
Comment (rkapka) [reply validator/accounts/account.go]: My bad. I misunderstood how `defer` works. I'll need to do some more reading about this. It should be fixed now.

File: validator/accounts/account.go (hunk 2 above)
Comment (rkapka) [reply validator/accounts/account.go]: As above.

**Pull request merged**